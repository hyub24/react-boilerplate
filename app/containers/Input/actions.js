/*
 *
 * Input actions
 *
 */

import {
  HANDLE_INPUT,
  SUBMIT_INPUT,
  SUBMIT_INPUT_SUCCESS,
  SUBMIT_INPUT_ERROR,
} from './constants';

export function handleInput(input) {
  return {
    type: HANDLE_INPUT,
    input,
  };
}

export function submitInput() {
  return {
    type: SUBMIT_INPUT,
  };
}

export function inputSubmitted() {
  return {
    type: SUBMIT_INPUT_SUCCESS,
  };
}

export function inputSubmittingError(error) {
  return {
    type: SUBMIT_INPUT_ERROR,
    error,
  };
}
