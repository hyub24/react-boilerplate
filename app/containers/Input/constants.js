/*
 *
 * Input constants
 *
 */

export const HANDLE_INPUT = 'app/Input/HANDLE_INPUT';
export const SUBMIT_INPUT = 'app/Input/SUBMIT_INPUT';
