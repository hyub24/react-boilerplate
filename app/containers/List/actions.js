/*
 *
 * List actions
 *
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_ERROR,
  LOAD_STRINGS_SUCCESS,
} from './constants';

export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}

export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

export function stringsLoadingError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}
