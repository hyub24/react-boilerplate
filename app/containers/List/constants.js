/*
 *
 * List constants
 *
 */

export const LOAD_STRINGS = 'app/List/LOAD_STRINGS';
