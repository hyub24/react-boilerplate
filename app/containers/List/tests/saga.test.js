/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put, takeLatest } from 'redux-saga/effects';
import listSaga, { getStrings } from '../saga';
import { stringsLoaded } from '../actions';
import { LOAD_STRINGS } from '../constants';

describe('getStrings Saga', () => {
  const generator = getStrings();

  beforeEach(() => {
    const selectDescriptor = generator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the stringsLoaded action if the call is successful', () => {
    const strings = [
      {
        id: 1,
        input: 'hello',
      },
      {
        id: 2,
        input: 'goodbye',
      },
    ];
    const putDescriptor = generator.next(strings).value;
    expect(putDescriptor).toEqual(put(stringsLoaded(strings)));
  });
});

describe('listSaga', () => {
  it('should start task to watch for LOAD_STRINGS action', () => {
    const takeLatestDescriptor = listSaga().next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_STRINGS, getStrings));
  });
});
