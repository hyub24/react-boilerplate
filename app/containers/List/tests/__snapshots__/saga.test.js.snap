// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStrings Saga should dispatch the stringsLoaded action if the call is successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/inputs",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
